lib/mixpanel-ruby/groups.rb

```diff
@@ -0,0 +1,180 @@
+        '$set_once' => properties,
+      }.merge(optional_params)
+      message['$ip'] = ip if ip
+
+      update(message)
+    end
+
+    # Set union on list valued properties.
+    # Associates a list containing all elements of a given list,
+    # and all elements currently in a list associated with the given
+    # property. After a union, every element in the list associated
+    # with a property will be unique.
+    #
+    #    tracker = Mixpanel::Tracker.new(YOUR_MIXPANEL_TOKEN)
+    #    tracker.groups.union("GROUP KEY", "1234", {
+    #        'Levels Completed' => ['Suffragette City']
+    #    });
+    #
+    def union(group_key, group_id, properties, ip=nil, optional_params={})
+      properties = fix_property_dates(properties)
+      message = {
+        '$group_key' => group_key,
+        '$group_id' => group_id,
+        '$union' => properties,
+      }.merge(optional_params)
+      message['$ip'] = ip if ip
+
+      update(message)
+    end
+
+    # Removes properties and their values from a group.
+    #
+    #    tracker = Mixpanel::Tracker.new(YOUR_MIXPANEL_TOKEN)
+    #
+    #    # removes a single property and its value from a group
+    #    tracker.groups.unset("GROUP KEY", "1234", "Overdue Since")
+    #
+    #    # removes multiple properties and their values from a group
+    #    tracker.groups.unset("GROUP KEY",
+    #                         "1234",
+    #                         ["Overdue Since", "Paid Date"])
+    #
+    def unset(group_key, group_id, properties, ip=nil, optional_params={})
+      properties = [properties] unless properties.is_a?(Array)
+      message = {
+        '$group_key' => group_key,
+        '$group_id' => group_id,
+        '$unset' => properties,
+      }.merge(optional_params)
+      message['$ip'] = ip if ip
+
+      update(message)
+    end
+
+    # Permanently delete a group from \Mixpanel groups analytics
+    def delete_group(group_key, group_id, optional_params={})
+      update({
+        '$group_key' => group_key,
+        '$group_id' => group_id,
+        '$delete' => '',
+      }.merge(optional_params))
+    end
+
+    # Send a generic update to \Mixpanel groups analytics.
+    # Caller is responsible for formatting the update message, as
+    # documented in the \Mixpanel HTTP specification, and passing
+    # the message as a dict to #update. This
+    # method might be useful if you want to use very new
+    # or experimental features of groups analytics from Ruby
+    # The \Mixpanel HTTP tracking API is documented at
+    # https://mixpanel.com/help/reference/http
+    def update(message)
+      data = {
+        '$token' => @token,
+        '$time' =>  ((Time.now.to_f) * 1000.0).to_i,
+      }.merge(message)
+
+      message = {'data' => data}
+
+      ret = true
+      begin
+        @sink.call(:group_update, message.to_json)
+      rescue MixpanelError => e
+        @error_handler.handle(e)
+        ret = false
+      end
+
+      ret
+    end
+
+    private
+
```

Comment:
you're missing one update function: $remove, which removes a value from a list-valued property (if present). Spec is `groupUpdates.remove(property name, value)`.

Reply:
True! There it is. :wink: